donate/payments/sqs.py

```diff
@@ -18,14 +24,40 @@ def sqs_client():
         )
 
 
+def attempt_send_to_sqs(client, payload):
+    client.send_message(
+        QueueUrl=settings.BASKET_SQS_QUEUE_URL,
+        MessageBody=json.dumps(payload, cls=DjangoJSONEncoder, sort_keys=True),
+    )
+
+
 def send_to_sqs(payload):
     # If BASKET_SQS_QUEUE_URL is not configured, do nothing (djangorq is logging the payload).
     if settings.BASKET_SQS_QUEUE_URL:
         client = sqs_client()
         if client is None:
+            logger.error("Could not connect to SQS Client.")
             return
 
-        return client.send_message(
-            QueueUrl=settings.BASKET_SQS_QUEUE_URL,
-            MessageBody=json.dumps(payload, cls=DjangoJSONEncoder, sort_keys=True),
-        )
+    send_retries = 3
+    send_data_immediately = 0  # seconds
+    send_data_delay = 10  # seconds
+    schedule_priority = 1
+    call_args = (client, payload)
+
+    for attempt in range(send_retries):
+        if attempt == 0:
+            schedule.enter(send_data_immediately, schedule_priority, attempt_send_to_sqs, call_args)
+        else:
+            schedule.enter(send_data_delay, schedule_priority, attempt_send_to_sqs, call_args)
+        try:
+            # explicitly block, just in case the implicit behaviour changes in the future
+            schedule.run(blocking=True)
+            break
+        except botocore.exceptions.ClientError as err:
+            if attempt != 2:
+                logger.error(f"Error when sending data to SQS: {err.response['Error']['Message']}")
```

Comment:
I'm pretty sure this is safe, but `err.response['Error']['Message']` is always tricky: what happens when `Message` doesn't exist? Or one level up, `Error`? It'd be better to make sure these exist before we try to template them into the string.

Reply:
@danielfmiranda I'd still recommend making sure this is a safe attribute access